.github/workflows/cypress.yml

```diff
@@ -0,0 +1,77 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the action will run.
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+    branches: [ main ]
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  npm-v6:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Lockfile version 1 (< npm 6.x)
+        run: "grep '\"lockfileVersion\": 1' package-lock.json"
+
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      # Cache the npm install cache
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.npm
+            ~/.cache/Cypress
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      # Install using ONLY the package-lock.json
+      - name: Install dependencies
+        run: npm ci
+      - name: Run tests
+        run: npm test
+
+  cypress-test:
+    runs-on: ubuntu-latest
+    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      # Cache the npm install cache
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      # Install using ONLY the package-lock.json
+      - name: Install dependencies
+        run: npm ci
+      - name: Start testing
+        uses: cypress-io/github-action@v2
+        with:
+          start: npm start
+          wait-on: 'http://localhost:8080'
+          command: npm run cypress:run
```

Comment:
Any reason this isn't being added as just another job on the existing CI workflow in `.github/workflows/main.yml`? seems like you just copy pasted everything from there.

Reply:
to have the option to turn it off if need be, wanted to preserve the original workflow if something was wrong with this one. Also separates errors from cypress from other errors.

Reply:
That's fine. In that case, can you remove the duplicate `lint` and `test` runs from this file? Also rename the workflow to something like `Cypress CI`. Otherwise the name shows up the same in Actions and it becomes somewhat confusing.